executor/executor.go

```diff
@@ -0,0 +1,235 @@
+// Package executor implements a high level execution context with monitoring and
+// logging features.
+//
+// Please see Executor for more information.
+package executor
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"os/exec"
+	"syscall"
+	"time"
+
+	"github.com/Sirupsen/logrus"
+)
+
+// ExecResult is the result of a Stop() operation and contains various fields
+// related to the post-mortem state of the process such as output and exit
+// status.
+type ExecResult struct {
+	Stdout     string
+	Stderr     string
+	ExitStatus uint32
```

Comment:
no, that's deliberately a uint32 so it's easier to check. note that it has a String() and Stop() handles the *exec.ExitError appropriately.